### PR TITLE
v3rpc: fill lease header

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/5173

### CODE REVIEW

File: etcdserver/api/v3rpc/lease.go [hunk 1]
```diff
@@ -67,7 +72,8 @@ func (ls *LeaseServer) LeaseKeepAlive(stream pb.Lease_LeaseKeepAliveServer) erro
 			return err
 		}
 
-		resp := &pb.LeaseKeepAliveResponse{ID: req.ID, TTL: ttl}
+		resp := &pb.LeaseKeepAliveResponse{ID: req.ID, TTL: ttl, Header: &pb.ResponseHeader{}}
+		ls.hdr.fill(resp.Header)
```
Comment (heyitsanthony) [inline etcdserver/api/v3rpc/lease.go]: If the lease is revoked at `rev1`, does it matter that `resp.Header.Revision > rev1` is possible here?

File: etcdserver/api/v3rpc/lease.go [hunk 2]
```diff
@@ -25,27 +25,32 @@ import (
...
+	return &LeaseServer{le: s, hdr: newHeader(s)}
 }
 
 func (ls *LeaseServer) LeaseGrant(ctx context.Context, cr *pb.LeaseGrantRequest) (*pb.LeaseGrantResponse, error) {
 	resp, err := ls.le.LeaseGrant(ctx, cr)
 	if err == lease.ErrLeaseExists {
 		return nil, rpctypes.ErrLeaseExist
 	}
+	resp.Header = &pb.ResponseHeader{}
+	ls.hdr.fill(resp.Header)
```
Comment (heyitsanthony) [inline etcdserver/api/v3rpc/lease.go]: should this be the revision at the time of the grant going through raft?

**Comment by xiang90** [comment]
@heyitsanthony I guess the question is do we want the rev in lease to be strictly matched with the mvcc store? If yes, then we need to fill in the header when it goes through raft.

**Comment by heyitsanthony** [comment]
@xiang90 if the system is going through the trouble of filling out the header I think it should have the precise revision (otherwise, ignoring the RTT cost, the client might as well do a serialized Get to get the rev). Possibly receiving a grant or keepalive with a revision `>=` than its revoke revision seems broken to me.

**Comment by xiang90** [comment]
@heyitsanthony Fixed. PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**